pipeline-model-extensions/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/agent/DeclarativeAgent.java

```diff
@@ -104,4 +107,14 @@ public DeclarativeAgentDescriptor getDescriptor() {
         return (DeclarativeAgentDescriptor) super.getDescriptor();
     }
 
+    @Extension
+    public static class CheckoutScriptAllowlist extends GroovySourceFileAllowlist {
```

Comment:
We need this because `DeclarativeAgent` itself is abstract. Another option could be to hard-code this file in `WithScriptDescriptor`.